### PR TITLE
Define cloud billing event types and codes

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/cloud/issues/442

#### Description
- Define cloud billing event types and codes
- Exported auth.clientUsername so it can be used in enterprise when emitting cloud related events (https://github.com/gravitational/teleport.e/pull/239)

### CODE REVIEW

File: lib/events/codes.go [hunk 1]
```diff
@@ -294,6 +294,14 @@ const (
 	MFADeviceAddEventCode = "T1006I"
 	// MFADeviceDeleteEventCode is an event code for users deleting MFA devices.
 	MFADeviceDeleteEventCode = "T1007I"
+	// BillingCardCreateCode is emitted when a user creates a new credit card.
+	BillingCardCreateCode = "T1008I"
```
Comment (alex-kovoy) [inline lib/events/codes.go:298]: If you look at numbers, the `T10xxx` is used for user login events. The billing deserve own group. 

Lets use `TBL` prefix for that:

`TBL01I`
`TBL02I`

Comment (kimlisa) [reply lib/events/codes.go]: revised

File: api/types/events/events.proto [hunk 2]
```diff
@@ -1263,6 +1297,9 @@ message OneOf {
         events.SessionUpload SessionUpload = 43;
         events.MFADeviceAdd MFADeviceAdd = 44;
         events.MFADeviceDelete MFADeviceDelete = 45;
+        events.BillingAccountUpdate BillingAccountUpdate = 46;
```
Comment (alex-kovoy) [inline api/types/events/events.proto:1300]: `BillingAcountUpdate` -> `BillingInformationUpdate`

File: lib/events/api.go [hunk 3]
```diff
@@ -192,6 +192,15 @@ const (
...
 	AccessRequestID = "id"
 
+	// BillingCardCreateEvent is emitted when a user creates a new credit card.
+	BillingCardCreateEvent = "billing.create_card"
+	// BillingCardDeleteEvent is emitted when a user deletes a credit card.
+	BillingCardDeleteEvent = "billing.delete_card"
+	// BillingCardUpdateEvent is emitted when a user updates an existing credit card.
+	BillingCardUpdateEvent = "billing.update_card"
+	// BillingAccountUpdateEvent is emitted when a user updates their billing account.
+	BillingAccountUpdateEvent = "billing.update_account"
```
Comment (alex-kovoy) [inline lib/events/api.go:202]: BillingAccountUpdate -> BillingInformationUpdate

**Review by alex-kovoy — APPROVED** [review]
with comments

File: lib/events/api.go (hunk 3 above)
Comment (kimlisa) [reply lib/events/api.go]: revised

**Review by russjones — APPROVED** [review]
Bot.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**